packages/mdc-menu/index.js

```diff
@@ -27,79 +29,96 @@ class MDCMenu extends MDCComponent {
   }
 
   /**
-   * @param {!Element} root
+   * @param {!HTMLElement} root
    * @return {!MDCMenu}
    */
   static attachTo(root) {
     return new MDCMenu(root);
   }
 
+  initialize(
+    menuSurfaceFactory = (el) => new MDCMenuSurface(el),
+    listFactory = (el) => new MDCList(el)) {
+    this.menuSurface_ = menuSurfaceFactory(this.root_);
+    this.afterOpenedCallback_ = () => this.handleAfterOpened_();
+
+    const list = this.root_.querySelector(strings.LIST_SELECTOR);
+    if (list) {
+      this.list_ = listFactory(list);
+      this.list_.wrapFocus = true;
+    }
+    this.menuSurface_.listen(MDCMenuSurfaceFoundation.strings.OPENED_EVENT, this.afterOpenedCallback_);
+  }
+
+  destroy() {
+    if (this.list_) {
+      this.list_.destroy();
+    }
+
+    this.menuSurface_.destroy();
+    this.menuSurface_.unlisten(MDCMenuSurfaceFoundation.strings.OPENED_EVENT, this.afterOpenedCallback_);
+    this.deregisterListeners_();
+    super.destroy();
+  }
+
   /** @return {boolean} */
   get open() {
-    return this.foundation_.isOpen();
+    return this.menuSurface_.open;
   }
 
   /** @param {boolean} value */
   set open(value) {
     if (value) {
-      this.foundation_.open();
+      this.menuSurface_.open = true;
+      this.registerListeners_();
     } else {
-      this.foundation_.close();
+      this.deregisterListeners_();
+      this.menuSurface_.open = false;
     }
   }
 
-  /** @param {{focusIndex: ?number}=} options */
-  show({focusIndex = null} = {}) {
-    this.foundation_.open({focusIndex: focusIndex});
-  }
-
-  hide() {
-    this.foundation_.close();
-  }
-
   /**
-   * @param {Corner} corner Default anchor corner alignment of top-left
+   * @param {!Corner} corner Default anchor corner alignment of top-left
    *     menu corner.
    */
   setAnchorCorner(corner) {
-    this.foundation_.setAnchorCorner(corner);
+    this.menuSurface_.setAnchorCorner(corner);
   }
 
   /**
-   * @param {AnchorMargin} margin
+   * @param {!AnchorMargin} margin
    */
   setAnchorMargin(margin) {
-    this.foundation_.setAnchorMargin(margin);
-  }
-
-  /**
-   * Return the item container element inside the component.
-   * @return {?Element}
-   */
-  get itemsContainer_() {
-    return this.root_.querySelector(MDCMenuFoundation.strings.ITEMS_SELECTOR);
+    this.menuSurface_.setAnchorMargin(margin);
   }
 
   /**
    * Return the items within the menu. Note that this only contains the set of elements within
    * the items container that are proper list items, and not supplemental / presentational DOM
    * elements.
-   * @return {!Array<!Element>}
+   * @return {!Array<!HTMLElement>}
    */
   get items() {
-    const {itemsContainer_: itemsContainer} = this;
-    return [].slice.call(itemsContainer.querySelectorAll('.mdc-list-item[role]'));
+    return this.list_.listElements_;
   }
 
   /**
-   * Return the item within the menu that is selected.
+   * Return the item within the menu at the index specified.
```

Comment:
Do we foresee a purpose for this API? I know it existed before but I'm not sure why.

Reply:
I believe this was here previously so we could attempt to replicate `.item(index)` functionality from the native select. I planned on implementing this in select so I included it here, but it may not be required.